rasa/core/evaluation/markers.py

```diff
@@ -56,11 +56,11 @@ def from_file(cls, path: Text) -> Dict:
     def from_yaml(cls, yaml: Text, filename: Text = "") -> Dict:
         """Loads the config from YAML text after validating it."""
         try:
-            # TODO implement and call the yaml schema validation
-            #  rasa.shared.utils.validation.validate_yaml_schema(yaml,
-            #  MAKERS_SCHEMA_FILE)
 
-            return rasa.shared.utils.io.read_yaml(yaml)
+            config = rasa.shared.utils.io.read_yaml(yaml)
+            cls._validate_config(config)
+            return config
+
         except ParserError as e:
             e.filename = filename
             rasa.shared.utils.io.raise_warning(
```

Comment:
Not directly related to this PR, but here I would actually suggest raising a `RasaException` since this would be a usage error (and to prevent a tsunami of Sentry errors because of invalid Yaml):
```suggestion
            raise RasaException(message=...) from e
```

Reply:
I actually found a more useful subclass of `RasaException` [here.](https://github.com/RasaHQ/rasa/blob/0795099f5b237e648aed2e46397643f891da0378/rasa/shared/exceptions.py#L38)